src/MoBi.Presentation/Tasks/SimulationCommitTask.cs

```diff
@@ -0,0 +1,126 @@
+         var templateModule = _templateResolverTask.TemplateModuleFor(lastModuleConfiguration.Module);
+         
+         var moBiMacroCommand = new MoBiMacroCommand
+         {
+            CommandType = CommitCommand,
+            Description = CommitCommandDescription(templateModule, simulationWithChanges),
+            ObjectType = new ObjectTypeResolver().TypeFor<Module>()
+         };
+         moBiMacroCommand.Add(lastModuleConfiguration.SelectedInitialConditions == null ? 
+            addNewInitialConditionsFromSimulationChanges(simulationWithChanges, templateModule) : 
+            updateInitialConditionsFromSimulationChanges(simulationWithChanges, lastModuleConfiguration.SelectedInitialConditions));
+
+         moBiMacroCommand.Add(lastModuleConfiguration.SelectedParameterValues == null ? 
+            addNewParameterValuesFromSimulationChanges(simulationWithChanges, templateModule) : 
+            updateParameterValuesFromSimulationChanges(simulationWithChanges, lastModuleConfiguration.SelectedParameterValues));
+
+         return moBiMacroCommand.Run(_context);
+      }
+
+      private ICommand updateParameterValuesFromSimulationChanges(IMoBiSimulation simulationWithChanges, ParameterValuesBuildingBlock selectedParameterValues)
+      {
+         return new MoBiEmptyCommand();
+      }
+
+      private ICommand updateInitialConditionsFromSimulationChanges(IMoBiSimulation simulationWithChanges, InitialConditionsBuildingBlock selectedInitialConditions)
+      {
+         return new MoBiEmptyCommand();
+      }
+
+      private IMoBiCommand addNewParameterValuesFromSimulationChanges(IMoBiSimulation simulationWithChanges, Module lastModule)
+      {
+         var parameterValuesToAdd = changesFrom<Parameter>(simulationWithChanges).Select(x => createParameterValue(x.quantity, x.objectPath)).ToList();
+         return createAddBuildingBlockCommand<ParameterValuesBuildingBlock, ParameterValue>(simulationWithChanges.Name, lastModule.ParameterValuesCollection.AllNames(), lastModule, parameterValuesToAdd);
+      }
+
+      private IMoBiCommand addNewInitialConditionsFromSimulationChanges(IMoBiSimulation simulationWithChanges, Module lastModule)
+      {
+         var initialConditionsToAdd = changesFrom<MoleculeAmount>(simulationWithChanges).Select(x => createInitialCondition(x.quantity, x.objectPath)).ToList();
+         return createAddBuildingBlockCommand<InitialConditionsBuildingBlock, InitialCondition>(simulationWithChanges.Name, lastModule.InitialConditionsCollection.AllNames(), lastModule, initialConditionsToAdd);
+      }
+
+      /// <summary>
+      ///    Returns path and quantity tuple for all changes in the <paramref name="simulationWithChanges" /> of type
+      ///    <typeparamref name="TQuantity" />
+      /// </summary>
+      private IEnumerable<(ObjectPath objectPath, TQuantity quantity)> changesFrom<TQuantity>(IMoBiSimulation simulationWithChanges) where TQuantity : Quantity
```

Comment:
rename objectPath => quantityPath. Otherwise it's not clear that the belong together